c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -482,6 +482,41 @@ class CachingAllocatorConfig {
   }
 };
 
+// records allocate/free events
+class MemoryEventTracker {
+ private:
+  std::mutex mutex;
+
+ public:
+  std::vector<std::vector<AllocFreeEvent>> alloc_free_events;
+  std::vector<unsigned int> num_alloc_events;
+  void append_alloc_free_event(
+      intptr_t ptr,
+      size_t size,
+      bool type,
+      int device) {
+    std::lock_guard<std::mutex> lock(mutex);
+    if (num_alloc_events[device] >= alloc_free_events[device].size()) {
+      alloc_free_events[device].resize(alloc_free_events[device].size() * 2);
```

Comment:
why do you need manually resize vector instead of letting vector handle it automatically?

Reply:
I noticed that the number of events can large. If the vector is extended by one element each iteration and the running time becomes quadratic, this can be slow. I double the size when we need more space to maintain a linear amortized running time.

Reply:
I tested this and there is no big difference in the running time. I let the vector handle resizing now. Thank you.